docs/model_developers/release.rst

```diff
@@ -57,23 +78,46 @@ Or for the current project
 
-Distributing Inmanta modules
-############################
+Distributing V1 modules
+-----------------------
 
-Inmanta modules can be distributed in two different ways: using a Git repository (V1 modules) or using a Python package (V2
-modules).
+V1 modules are generally simply distributed using a Git repository. They can however also be built as a V2 Python package
+and distributed the same as other V2 modules.
 
 Git repository distribution format
-----------------------------------
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 Distributing a V1 module using a Git repository happens by storing the source code of that module on a Git repository
 that is accessible by the Inmanta orchestrator. The orchestrator will clone the source code of the module and install it in the
-Inmanta project.
-
-Python package distribution format
-----------------------------------
-
-Modules defined in the V2 module format can be distributed as a Python package. Run the ``inmanta module build`` command in
-the source directory of a module to build a Python Wheel from that module. The resulting package is stored in the dist directory
-of the module. The Python packages should be stored on a Python package repository, reachable by the orchestrator.
-Uploading packages to the package repository should be done with the appropriate tool for the specific repository at hand.
+Inmanta project. Tagging release versions as outlined above allows specifying constraints on the module version.
+
+V2 package distribution format
+------------------------------
+
+A V2 package can be built for a V1 module with ``inmanta module build``. This package can be distributed as described in
+:ref:`modules-distribution-v2`.
+Modules installed from a package will always act as V2 modules and will be considered such by the compiler.
+
+
+.. _modules-distribution-advanced-concepts:
+
+Advanced concepts
+#################
+
+Freezing a project
+------------------
+Prior to releasing a new stable version of an inmanta project, you might wish to freeze its module
+dependencies. This will ensure that the orchestrator server will always work with the exact
+versions specified. You can achieve this with
+``inmanta project freeze --recursive --operator "=="``. This command will freeze all module
+dependencies to their exact version as they currently exist in the Python environment. The recursive
+option makes sure all module dependencies are frozen, not just the direct dependencies. In other
+words, if the project depends on module ``a`` which in turn depends on module ``b``, both modules
+will be pinned to their current version in ``setup.cfg``.
+
+Manual export
+-------------
+The ``inmanta export`` command exports a project and all its modules' ``inmanta_plugins`` packages
+to the orchestrator server. When this method is used, the orchestrator does not install any modules
+from the Python package repository but instead contains all Python code as present in the local
```

Comment:
I think inmanta export shouldn't be mentioned when talking about distribution, as it has nothing to do with that. 

I think this will create undue confusion.

Reply:
Right, I'll give this some more thought. I don't want to just get rid of it because it does serve to highlight an exception to how the server installs modules.

Reply:
I've moved this as well as the part referencing it back to the modules installation section under a "Module installation on the server" subsection. I've promoted the freezing a project section. Let me know what you think.